### PR TITLE
Adds Tooltip to Operation in Pallet. Shortens description.

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

File: src/main/java/edu/wpi/grip/ui/OperationView.java [hunk 1]
```diff
@@ -59,16 +62,29 @@ public OperationView(EventBus eventBus, Operation operation) {
 
     @Override
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        final double iconSizeDpi = ICON_SIZE_INCHES * Screen.getPrimary().getDpi();
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: `iconSizePixels` would be a more accurate name I think

File: src/main/java/edu/wpi/grip/ui/OperationView.java [hunk 2]
```diff
@@ -59,16 +62,29 @@ public OperationView(EventBus eventBus, Operation operation) {
...
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        final double iconSizePixels = ICON_SIZE_INCHES * Screen.getPrimary().getDpi();
+
         this.setId(StyleClassNameUtility.idNameFor(this.operation));
         this.name.setText(this.operation.getName());
         this.description.setText(this.operation.getDescription());
+        this.description.setTextOverrun(OverrunStyle.WORD_ELLIPSIS);
+        this.description.setMaxHeight(iconSizePixels);
+
+        final Tooltip tooltip = new Tooltip(this.operation.getDescription());
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: Don't add a tooltip if the description is empty

File: src/main/java/edu/wpi/grip/ui/OperationView.java [hunk 3]
```diff
@@ -59,16 +62,29 @@ public OperationView(EventBus eventBus, Operation operation) {
...
+        tooltip.setPrefWidth(400.0);
+        tooltip.setWrapText(true);
+        tooltip.setTextAlignment(TextAlignment.JUSTIFY);
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: Left justify would look better
justify

File: src/main/java/edu/wpi/grip/ui/OperationView.java [hunk 4]
```diff
@@ -59,16 +62,29 @@ public OperationView(EventBus eventBus, Operation operation) {
 
     @Override
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        final double iconSizePixels = ICON_SIZE_INCHES * Screen.getPrimary().getDpi();
+
         this.setId(StyleClassNameUtility.idNameFor(this.operation));
         this.name.setText(this.operation.getName());
         this.description.setText(this.operation.getDescription());
+        this.description.setTextOverrun(OverrunStyle.WORD_ELLIPSIS);
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: This should probably be set from the FXML

File: src/main/java/edu/wpi/grip/ui/OperationView.java (hunk 2 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: Why don't we make the description an optional field object?

File: src/main/java/edu/wpi/grip/ui/OperationView.java (hunk 4 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: How?

File: src/main/java/edu/wpi/grip/ui/OperationView.java (hunk 3 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: Okay.

File: src/main/java/edu/wpi/grip/ui/OperationView.java (hunk 4 above)
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: `<Label fx:id="description"` ... `textOverrun="WORD_ELLIPSIS"` ... `/>`

File: src/main/java/edu/wpi/grip/ui/OperationView.java (hunk 2 above)
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: we could do that

Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/ui/OperationView.java]: Want that in a separate PR?

**Comment by JLLeitschuh** [comment]
Tom
> Don't add a tooltip if the description is empty

Jonathan
> Why don't we make the description an optional field object?

Tom
> we could do that

Jonathan
> Want that in a separate PR?

**Pull request merged**